src/d3d12/d3d12_resource_pool_texture.cpp

```diff
@@ -90,6 +111,109 @@ namespace wr
 		}
 	}
 
+	void D3D12TexturePool::Stage()
+	{
+		size_t unstaged_number = m_unstaged_textures.size();
+
+		if (unstaged_number > 0)
+		{
+			if (m_is_staging)
+			{
+				d3d12::WaitFor(m_staging_fence);
+				m_is_staging = false;
+			}
+
+			std::vector<d3d12::TextureResource*> unstaged_textures;
+
+			auto itr = m_unstaged_textures.begin();
+
+			for (itr; itr != m_unstaged_textures.end(); ++itr)
+			{
+				unstaged_textures.push_back(static_cast<d3d12::TextureResource*>(itr->second));
+			}
+
+			for (int i = 0; i < d3d12::settings::max_threads; ++i) {
+
+				d3d12::CommandList* cmdlist = m_command_lists[i];
+
+				m_command_recording_futures[i] = m_thread_pool->Enqueue([i](
```

Comment:
This code can be greatly simplified if you use the `DivideWork` function from the thread pool. As I mentioned when I gave you this task. See the frame graph for how it works.

Reply:
That would have been nice to have some documentation on, since I've no clue on how DivideWork works, and I can't remember being told the frame graph has an example. Also, wasn't that code bugged? I think Emilio found some weird error in it.